src/umockcallrecorder.c

```diff
@@ -475,107 +604,140 @@ UMOCKCALLRECORDER_HANDLE umockcallrecorder_clone(UMOCKCALLRECORDER_HANDLE umock_
     }
     else
     {
-        /* Codes_SRS_UMOCKCALLRECORDER_01_037: [ If allocating memory for the new umock call recorder instance fails, umockcallrecorder_clone shall fail and return NULL. ]*/
-        result = umockcallrecorder_create();
-        if (result != NULL)
+        /* Codes_SRS_UMOCKCALLRECORDER_01_082: [ If a lock was created for the call recorder umock_call_recorder, umockcallrecorder_clone shall acquire_the lock in shared mode for umock_call_recorder. ]*/
+        internal_lock_acquire_shared_if_needed(umock_call_recorder);
         {
-            size_t i;
-
-            result->expected_calls = (UMOCK_EXPECTED_CALL*)umockalloc_malloc(sizeof(UMOCK_EXPECTED_CALL) * umock_call_recorder->expected_call_count);
-            if (result->expected_calls == NULL)
+            /* Codes_SRS_UMOCKCALLRECORDER_01_037: [ If allocating memory for the new umock call recorder instance fails, umockcallrecorder_clone shall fail and return NULL. ]*/
+            result = umockcallrecorder_create(NULL, NULL);
```

Comment:
>NULL, NUL [](start = 46, length = 9)

why not just pass the factory in here? Shouldn't that save some lines below?

Reply:
Changed, thank you!

---
In reply to: [530654785](https://github.com/Azure/umock-c/pull/177#discussion_r530654785) [](ancestors = 530654785)